integration-cli/docker_cli_build_test.go

```diff
@@ -6488,3 +6488,19 @@ RUN touch /foop
 	c.Assert(err, check.IsNil)
 	c.Assert(d.String(), checker.Equals, getIDByName(c, name))
 }
+
+// Test case for #34208
+func (s *DockerSuite) TestBuildAddHTTPRoot(c *check.C) {
+	testRequires(c, Network, DaemonIsLinux)
+	buildImageSuccessfully(c, "buildaddhttproot", build.WithDockerfile(`
+                FROM scratch
+                ADD http://example.com/index.html /example1
+                ADD http://example.com /example2
+                ADD http://example.com /example3`))
```

Comment:
This shouldn't be a slow integration test. Any time `example.com` is unavailable the build will fail, which is not good.

This bug was in a single small function. We can test this with a fast unit test. Splitting out `getTargetFilename()` will make it even easier to test because we won't even need to perform an HTTP request to exercise the bug.